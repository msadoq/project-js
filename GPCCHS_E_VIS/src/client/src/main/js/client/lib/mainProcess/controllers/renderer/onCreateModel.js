import { dirname } from 'path';
import { getStore } from '../../../store/mainStore';
import { getView } from '../../../store/selectors/views';
import { add } from '../../../store/actions/messages';
import { getPathByFilePicker } from '../../dialog';
import { saveViewAs } from '../../../documentManager';
import { getRootDir } from '../../../common/fmd';
import { getViewModule } from '../../../viewManager';

const root = getRootDir();

export default function ({ viewId }) {
  const { getState, dispatch } = getStore();
  const { type, configuration, absolutePath } = getView(getState(), { viewId });

  const folder = absolutePath ? dirname(absolutePath) : root;
  const modelConfiguration = getViewModule(type).prepareConfigurationForModel(configuration);
  return getPathByFilePicker(folder, 'model', 'save', (err, path) => {
    saveViewAs(modelConfiguration, type, path, (errSaving) => {
      if (errSaving) {
        dispatch(add(viewId, 'danger', `Model unsaved ${errSaving}`));
      } else {
        dispatch(add(viewId, 'success', 'Model saved'));
      }
    });
  });
}
