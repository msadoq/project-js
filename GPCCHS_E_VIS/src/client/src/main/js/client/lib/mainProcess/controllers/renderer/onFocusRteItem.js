import asyncParallel from 'async/parallel';
import { getStore } from 'mainProcess/store';
import _reduce from 'lodash/reduce';
import _set from 'lodash/set';
import _indexOf from 'lodash/indexOf';
import {
  setRteCatalogs,
  setRteDomains,
  setRteItemNames,
  setRteFocusedItem,
} from 'store/actions/rte';
import { getRtd } from 'rtdManager';
import prepareDataToTree from 'rtdManager/prepareDataToTree';
import { add } from 'store/actions/messages';
import {
  getRteFocusedInfo,
} from 'store/reducers/rte';
import asyncMap from 'async/map';

export default function ({ sessionId, domainId, catalog, version, namespace, name, key }) {
  const { getState, dispatch } = getStore();

  const state = getState();
  const focusedInfo = getRteFocusedInfo(state);

  const rtd = getRtd();
  const requests = [];

  if (sessionId !== focusedInfo.session) {
    // focus domain, request catalogs
    requests.push((callback) => {
      rtd.getDatabase().getDomainListBySession(sessionId, (err, domains) => {
        if (err) {
          callback(err);
          return;
        }
        dispatch(setRteDomains(sessionId, domains));
        callback(null);
      });
    });
  }
  if (domainId !== focusedInfo.domain) {
    // focus session, request domains
    requests.push((callback) => {
      rtd.getDatabase().getCatalogAndVersionListByDomain(
        sessionId, domainId, (err, catalogList) => {
          if (err) {
            callback(err);
            return;
          }
          const catalogs = prepareDataToTree(catalogList);
          const catalogIdx = _indexOf(Object.keys(catalogList), catalog);
          _set(catalogs, ['children', catalogIdx, 'toggled'], true);
          _set(catalogs, ['children', catalogIdx, 'active'], true);
          dispatch(setRteCatalogs(sessionId, domainId, catalogs));
          callback(null);
        }
      );
    });
  }
  if (catalog !== focusedInfo.catalog || version !== focusedInfo.version ||
      namespace !== focusedInfo.namespace || name !== focusedInfo.name) {
    // focus catalog, request items
    requests.push((callback) => {
      rtd.getDatabase().getNamespaceList(catalog, version, (err, namespaceList) => {
        if (err) {
          callback(err);
          return;
        }
        asyncMap(namespaceList, (ns, done) => {
          rtd.getDatabase().getItemList(catalog, version, ns, (iErr, itemList) => {
            if (iErr) {
              done(iErr);
              return;
            }
            done(null, { ns, itemList });
          });
        }, (aErr, res) => {
          if (aErr) {
            callback(aErr);
            return;
          }
          const itemMap = _reduce(res, (previous, value) => {
            if (!value) {
              return previous;
            }
            const item = Object.assign({}, previous, {
              [value.ns]: value.itemList,
            });
            return item;
          }, {});
          const items = prepareDataToTree(itemMap);
          const namespaceIdx = _indexOf(Object.keys(itemMap), namespace);
          const nameIdx = _indexOf(itemMap[namespace], name);
          _set(items, ['children', namespaceIdx, 'toggled'], true);
          _set(items, ['children', namespaceIdx, 'children', nameIdx, 'active'], true);
          dispatch(setRteItemNames(catalog, version, items));
          callback(null);
        });
      });
    });
  }

  asyncParallel(requests, (err) => {
    if (err) {
      dispatch(add('global', 'warning', err));
      return;
    }
    dispatch(setRteFocusedItem(key, sessionId, domainId, catalog, version, namespace, name));
  });
}
