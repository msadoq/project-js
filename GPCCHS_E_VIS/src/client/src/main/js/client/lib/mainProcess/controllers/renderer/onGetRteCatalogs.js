// ====================================================================
// HISTORY
// VERSION : 1.1.2 : DM : #6688 : 05/07/2017 : update catalog explorer controllers with new rtd connection
// VERSION : 1.1.2 : DM : #6688 : 05/07/2017 : catalog explorer : open, close and browse items
// VERSION : 1.1.2 : DM : #6688 : 05/07/2017 : First draft on catalog explorer
// END-HISTORY
// ====================================================================

import { getStore } from '../../store';
import { getRtd } from '../../../rtdManager';
import prepareDataToTree from '../../../rtdManager/prepareDataToTree';
import { add } from '../../../store/actions/messages';
import {
  setRteCatalogs,
} from '../../../store/actions/rte';


export default function ({ sessionId, domainId }) {
  const { dispatch } = getStore();

  const rtd = getRtd();
  rtd.getDatabase().getCatalogAndVersionListByDomain(sessionId, domainId, (err, catalogList) => {
    if (err) {
      dispatch(add('global', 'warning', err));
      return;
    }
    const catalogs = prepareDataToTree(catalogList);
    dispatch(setRteCatalogs(sessionId, domainId, catalogs));
  });
}
