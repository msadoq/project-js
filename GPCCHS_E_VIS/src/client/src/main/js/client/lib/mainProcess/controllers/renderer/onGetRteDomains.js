import { getStore } from 'mainProcess/store';
import { add } from 'store/actions/messages';
import { getRtd } from 'rtdManager';
import {
  setRteDomains,
} from 'store/actions/rte';


export default function ({ sessionId }) {
  const { dispatch } = getStore();

  const rtd = getRtd();
  rtd.getDatabase().getDomainListBySession(sessionId, (err, domains) => {
    if (err) {
      dispatch(add('global', 'warning', err));
      return;
    }
    dispatch(setRteDomains(sessionId, domains));
  });
}
