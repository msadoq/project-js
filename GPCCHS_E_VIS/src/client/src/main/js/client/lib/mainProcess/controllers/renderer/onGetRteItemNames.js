// ====================================================================
// HISTORY
// VERSION : 1.1.2 : DM : #6688 : 05/07/2017 : update catalog explorer controllers with new rtd connection
// VERSION : 1.1.2 : DM : #6688 : 05/07/2017 : catalog explorer : open, close and browse items
// VERSION : 1.1.2 : DM : #6688 : 05/07/2017 : First draft on catalog explorer
// END-HISTORY
// ====================================================================

import asyncMap from 'async/map';
import _reduce from 'lodash/reduce';
import { getStore } from '../../store';
import { getRtd } from '../../../rtdManager';
import prepareDataToTree from '../../../rtdManager/prepareDataToTree';
import { add } from '../../../store/actions/messages';
import {
  setRteItemNames,
} from '../../../store/actions/rte';


export default function ({ catalog, version }) {
  const { dispatch } = getStore();

  const rtd = getRtd();
  rtd.getDatabase().getNamespaceList(catalog, version, (err, namespaceList) => {
    if (err) {
      dispatch(add('global', 'warning', err));
      return;
    }

    asyncMap(namespaceList, (namespace, done) => {
      rtd.getDatabase().getItemList(catalog, version, namespace, (iErr, itemList) => {
        if (iErr) {
          done(iErr);
          return;
        }
        done(null, { namespace, itemList });
      });
    }, (aErr, res) => {
      if (aErr) {
        dispatch(add('global', 'warning', err));
        return;
      }
      const itemMap = _reduce(res, (previous, value) => {
        if (!value) {
          return previous;
        }
        const item = Object.assign({}, previous, {
          [value.namespace]: value.itemList,
        });
        return item;
      }, {});
      const items = prepareDataToTree(itemMap);
      dispatch(setRteItemNames(catalog, version, items));
    });
  });
}
