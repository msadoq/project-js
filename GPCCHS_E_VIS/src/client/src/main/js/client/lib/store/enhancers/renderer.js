/**
 * Renderer process store enhancer
 *
 * - intercepts dispatched actions => forwards to main process
 * - receives state diff from main process => apply
 */

export default function rendererStoreEnhancer(previousStoreCreator) {
  if (process.type !== 'renderer') {
    throw new Error('rendererStoreEnhancer used in a non-renderer process');
  }

  return function createStore(reducer, initialState) {
    const store = previousStoreCreator(reducer, initialState);

    // TODO : enhance
    const storeDotDispatch = store.dispatch;
    store.dispatch = (action) => {
      console.log('VIMA ACTION on rendererStoreEnhancer');
      storeDotDispatch(action);
    };

    return store;
  };
}
