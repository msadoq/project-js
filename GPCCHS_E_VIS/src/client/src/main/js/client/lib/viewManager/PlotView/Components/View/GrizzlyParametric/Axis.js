import React, { Component, PropTypes } from 'react';
import classnames from 'classnames';
import styles from './GrizzlyChart.css';
import { lineType, labelStyleType } from './types';

const { shape, string, func, bool, arrayOf, oneOf, number } = PropTypes;

export default class Axis extends Component {
  static propTypes = {
    direction: oneOf(['horizontal', 'vertical']).isRequired,
    lines: arrayOf(lineType.isRequired).isRequired,
    // showLabels: bool,
    label: string.isRequired,
    height: number.isRequired,
    xAxisHeight: number.isRequired,
    index: number.isRequired,
    showGrid: bool,
    chartWidth: number.isRequired,
    yAxisWidth: number.isRequired,
    labelStyle: shape(labelStyleType.isRequired),
    margin: number.isRequired,
    yAxesAt: string,
    xAxesAt: string,
    assignLabelEl: func.isRequired,
    assignEl: func.isRequired,
    memoizeAssignRef: func.isRequired,
    side: number.isRequired,
  };

  static defaultProps = {
    direction: 'horizontal',
    yAxesAt: 'left',
    xAxesAt: 'bottom',
    // showLabels: false,
    showGrid: true,
    labelStyle: {
      color: '#333333',
      bgColor: '#FFFFFF',
      align: 'center',
      font: 'Arial',
      italic: false,
      bold: false,
      underline: false,
      size: 11,
    },
  };

  render() {
    const {
      direction,
      lines,
      // showLabels,
      label,
      height,
      xAxisHeight,
      index,
      showGrid,
      chartWidth,
      yAxisWidth,
      labelStyle,
      margin,
      yAxesAt,
      xAxesAt,
      side,
      assignLabelEl,
      assignEl,
      memoizeAssignRef,
    } = this.props;

    const divStyle = {};
    const style = {};
    let divClassName;
    let axisClassName;
    let axisLabel;
    let lineLabel;
    if (direction === 'vertical') {
      const axisWidth = index === 0 && showGrid ? chartWidth + yAxisWidth : yAxisWidth;

      divStyle.height = height;
      divStyle.width = axisWidth;
      // divStyle.top = top;
      if (yAxesAt === 'left') {
        divStyle.left = margin;
      } else if (yAxesAt === 'right' && index === 0) {
        divStyle.left = 0;
      } else if (yAxesAt === 'right') {
        divStyle.right = margin;
      } else {
        divStyle.left = 0;
      }

      // vertical position
      style.width = axisWidth;
      style.height = height;

      divClassName = styles.yAxisDiv;
      axisLabel = styles.yAxisLabel;
      lineLabel = styles.yAxisLineLabel;
      axisClassName = styles.yAxis;
    } else {
      divStyle.height = index === 0 && showGrid ? height + xAxisHeight : xAxisHeight;
      divStyle.width = chartWidth;
      if (xAxesAt === 'top') {
        divStyle.top = margin;
      } else {
        divStyle.top = index === 0 ? 0 : height + margin;
      }

      if (yAxesAt === 'left') {
        divStyle.left = side;
      } else if (yAxesAt === 'right') {
        divStyle.right = side;
      } else {
        divStyle.left = 0;
      }

      // vertical position
      style.width = chartWidth;
      style.height = xAxisHeight;

      divClassName = styles.xAxisDiv;
      axisLabel = styles.xAxisLabel;
      lineLabel = styles.xAxisLineLabel;
      axisClassName = styles.xAxis;
    }
    return (
      <div
        style={divStyle}
        className={divClassName}
      >
        <span
          ref={assignLabelEl}
          className={classnames(
            'label',
            axisLabel,
            {
              [styles.labelUnderline]: labelStyle.underline,
              [styles.labelBold]: labelStyle.bold,
              [styles.labelItalic]: labelStyle.italic,
            }
          )}
        >
          { label }
        </span>
        {
          lines.map(line =>
            <span
              key={line.id}
              className={classnames(
                'label',
                lineLabel,
                { hidden: !line.displayLine }
              )}
              ref={memoizeAssignRef(line.id)}
            >
              {line.id} ({line.unit})
            </span>
          )
        }
        <svg
          style={style}
          ref={assignEl}
          className={axisClassName}
        />
      </div>
    );
  }
}
