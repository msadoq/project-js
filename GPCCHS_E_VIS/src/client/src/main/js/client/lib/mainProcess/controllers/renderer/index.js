// ====================================================================
// HISTORY
// VERSION : 1.1.0 : : : 28/02/2017 : Initial version
// VERSION : 1.1.2 : DM : #3622 : 07/03/2017 : first draft on inspector: retrieve data from rtd on right-click
// VERSION : 1.1.2 : DM : #5822 : 16/03/2017 : resolve a rtd link in the inspector
// VERSION : 1.1.2 : DM : #5828 : 30/03/2017 : Add a F1 button in VIMA to open the docu wiki helper
// VERSION : 1.1.2 : DM : #5822 : 03/04/2017 : merge dev in working branch
// VERSION : 1.1.2 : DM : #5828 : 23/05/2017 : Move common/callback and common/ipc modules in client sub-component
// VERSION : 1.1.2 : DM : #6785 : 12/06/2017 : activate links in views .
// VERSION : 1.1.2 : DM : #5828 : 13/06/2017 : Move common/constants/ in client/ folder
// VERSION : 1.1.2 : DM : #5828 : 14/06/2017 : Cleanup client/ file organization and test helper modules
// VERSION : 1.1.2 : FA : ISIS-FT-2132 : 15/06/2017 : Ask to save before closing view or page
// VERSION : 1.1.2 : DM : #6700 : 16/06/2017 : Add store enhancers helpers code coverage and merge with dev
// VERSION : 1.1.2 : DM : #6700 : 26/06/2017 : Change windowProcess health management to dispatch a Redux action directly.
// VERSION : 1.1.2 : DM : #6785 : 29/06/2017 : Fix opening view link in a new page and read only path for link definition
// VERSION : 1.1.2 : DM : #6688 : 05/07/2017 : First draft on catalog explorer
// VERSION : 1.1.2 : DM : #6688 : 05/07/2017 : catalog explorer : open, close and browse items
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 06/07/2017 : Remove unused onSavePage ipc renderer to mainProcess controllers
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 06/07/2017 : Replace ipc openPage by askOpenPage redux action
// VERSION : 1.1.2 : FA : #7235 : 18/07/2017 : Clean old code in menuManager/IPC controller
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 18/07/2017 : Fix common/fmd, disable Links feature for now
// VERSION : 1.1.2 : DM : #6700 : 20/07/2017 : Remove obsolete onServerDebug interface, ipcs and model
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 21/07/2017 : Clean IPC about openInspector .
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 21/07/2017 : Clean IPC about FMD_GET/FMD_CREATE and RELOAD_SESSIONS/GET_SESSION_TIME
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 21/07/2017 : Clean IPC about documents .
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 21/07/2017 : Remove serverDebug ipc . .
// VERSION : 1.1.2 : DM : #6700 : 03/08/2017 : Merge branch 'dev' into dbrugne-data
// VERSION : 1.1.2 : FA : #7145 : 04/08/2017 : Clean IPC about opening wiki helper + create a store folder in mainProcess
// END-HISTORY
// ====================================================================

import constants from 'constants';

import handle from 'common/ipc/handle';
import reply from 'common/ipc/reply';

import onReduxCurrentState from './onReduxCurrentState';
import onReduxDispatch from './onReduxDispatch';
import onResolveLink from './onResolveLink';
import onResolveRteLink from './onResolveRteLink';
import onGetRteCatalogs from './onGetRteCatalogs';
import onGetRteItemNames from './onGetRteItemNames';
import onOpenRteItem from './onOpenRteItem';
import onFocusRteItem from './onFocusRteItem';
import onGetRteDomains from './onGetRteDomains';

const controller = {
  [constants.IPC_METHOD_REDUX_CURRENT_STATE]: (...args) => onReduxCurrentState(reply, ...args),
  [constants.IPC_METHOD_REDUX_DISPATCH]: onReduxDispatch,
  [constants.IPC_METHOD_RESOLVE_LINK]: onResolveLink,
  [constants.IPC_METHOD_GET_RTE_DOMAINS]: onGetRteDomains,
  [constants.IPC_METHOD_GET_RTE_CATALOGS]: onGetRteCatalogs,
  [constants.IPC_METHOD_GET_RTE_ITEM_NAMES]: onGetRteItemNames,
  [constants.IPC_METHOD_OPEN_RTE_ITEM]: onOpenRteItem,
  [constants.IPC_METHOD_FOCUS_RTE_ITEM]: onFocusRteItem,
  [constants.IPC_METHOD_RESOLVE_RTE_LINK]: onResolveRteLink,
};

export default (electronEvent, data) => handle(
  controller,
  data,
  response => electronEvent.sender.send('global', response)
);
