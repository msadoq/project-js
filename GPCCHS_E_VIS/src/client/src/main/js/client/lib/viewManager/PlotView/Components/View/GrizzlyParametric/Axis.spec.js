import React from 'react';
import renderer from 'react-test-renderer';
import { scaleLinear } from 'd3-scale';
import Axis from './Axis';

describe('Axis :: render', () => {
  const propsStub = {
    assignEl: () => null,
    assignLabelEl: () => null,
    chartWidth: 1000,
    direction: 'vertical',
    height: 1000,
    index: 0,
    label: 'VBat',
    labelStyle: {
      color: '#333333',
      bgColor: '#FFFFFF',
      align: 'center',
      font: 'Arial',
      italic: false,
      bold: false,
      underline: false,
      size: 11,
    },
    lines: [
      {
        colorAccessor: 'color',
        data: {
          1509985461000: {
            color: '#000000',
            masterTime: 1509985461000,
            refTime: 1509985461000,
            symbol: '131',
            valX: 1509985461000,
            value: 10,
            x: 10,
          },
          1509985462000: {
            color: '#F00000',
            masterTime: 1509985462000,
            refTime: 1509985462000,
            symbol: '132',
            valX: 1509985462000,
            value: 20,
            x: 20,
          },
          1509985463000: {
            color: '#FF0000',
            masterTime: 1509985463000,
            refTime: 1509985463000,
            symbol: '133',
            valX: 1509985463000,
            value: 30,
            x: 30,
          },
          1509985464000: {
            color: '#FFF000',
            masterTime: 1509985464000,
            refTime: 1509985464000,
            symbol: '134',
            valX: 1509985464000,
            value: 40,
            x: 40,
          },
        },
        dataAccessor: 'TMMGT_BC_VIRTCHAN3',
        fill: '#000000',
        id: 'TMMGT_BC_VIRTCHAN3',
        indexes: [1509985461000, 1509985462000, 1509985463000, 1509985464000],
        lineSize: 2,
        lineStyle: 'Continuous',
        pointSize: 10,
        pointStyle: 'Dot',
        tooltipFormatter: () => {},
        xAccessor: null,
        xAxisId: 'time',
        xAxis: {
          autoLimits: false,
          autoTick: true,
          calculatedExtents: [1510306876602, 1510307506602],
          extents: [1510306876602, 1510307506602],
          format: '.2f',
          formatAsDate: true,
          gridSize: 1,
          gridStyle: 'Continuous',
          id: 'time',
          label: 'time',
          labelStyle: {},
          orient: 'top',
          rank: 100,
          scale: scaleLinear()
            .domain([0, 100])
            .range([0, 1000]),
          showAxis: true,
          showGrid: true,
          // showLabels: true,
          showTicks: true,
          tickStep: 20000,
          unit: 'V',
        },
        xTooltipAccessor: null,
        yAccessor: null,
        yAxis: {
          autoLimits: false,
          autoTick: true,
          calculatedExtents: [130, 140],
          extents: [130, 140],
          format: '.3f',
          formatAsDate: false,
          gridSize: 1,
          gridStyle: 'Continuous',
          id: 'VBat',
          label: 'VBat',
          labelStyle: {
            align: 'left',
            bold: false,
            color: '#FFF',
            font: 'Arial',
            italic: false,
            size: 12,
            strikeOut: false,
            underline: false,
          },
          logSettings: {
            base: 10,
            max: 1000000000,
            min: 0.1,
          },
          logarithmic: false,
          orient: 'top',
          rank: 100,
          scale: scaleLinear()
            .domain([0, 150])
            .range([1000, 0]),
          showAxis: true,
          showGrid: true,
          // showLabels: true,
          showTicks: true,
          tickStep: 50,
          unit: 'V',
        },
        yAxisId: 'VBat',
        yTooltipAccessor: null,
      },
    ],
    margin: 0,
    memoizeAssignRef: () => null,
    showGrid: true,
    // showLabels: true,
    side: 90,
    xAxesAt: 'bottom',
    xAxisHeight: 40,
    yAxesAt: 'left',
    yAxisWidth: 90,
  };
  test('Axis :: render :: nominal case vertical', () => {
    const tree = renderer.create(
      <Axis {...propsStub} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Axis :: render :: nominal case horizontal', () => {
    const propsStub2 = {
      ...propsStub,
      direction: 'horizontal',
    };
    const tree = renderer.create(
      <Axis {...propsStub2} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
