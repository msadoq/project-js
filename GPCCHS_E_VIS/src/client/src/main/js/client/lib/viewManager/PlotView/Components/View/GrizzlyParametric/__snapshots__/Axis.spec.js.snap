// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Axis :: render Axis :: render :: nominal case horizontal 1`] = `
<div
  className="xAxisDiv"
  style={
    Object {
      "height": 1040,
      "left": 90,
      "top": 0,
      "width": 1000,
    }
  }
>
  <span
    className="label xAxisLabel"
  >
    VBat
  </span>
  <span
    className="label xAxisLineLabel"
  >
    TMMGT_BC_VIRTCHAN3
     (
    )
  </span>
  <svg
    className="xAxis"
    style={
      Object {
        "height": 40,
        "width": 1000,
      }
    }
  />
</div>
`;

exports[`Axis :: render Axis :: render :: nominal case vertical 1`] = `
<div
  className="yAxisDiv"
  style={
    Object {
      "height": 1000,
      "left": 0,
      "width": 1090,
    }
  }
>
  <span
    className="label yAxisLabel"
  >
    VBat
  </span>
  <span
    className="label yAxisLineLabel"
  >
    TMMGT_BC_VIRTCHAN3
     (
    )
  </span>
  <svg
    className="yAxis"
    style={
      Object {
        "height": 1000,
        "width": 1090,
      }
    }
  />
</div>
`;
